docs: update docs to reflect new PR targeting methods for release trains (#38401)
angular/angular: docs/TRIAGE_AND_LABELS.md

To communicate the target we use the following labels:

Targetting active release train:

* `PR target: major`: A breaking change that cannot be part of a minor or patch release
* `PR target: minor`: A new feature that cannot be part of a patch release
* `PR target: patch`: A bug fix, refactoring, documentation change, etc. that can be a part of a patch release

Special Cases:
* `PR target: rc`: A critical fix for an active release-train while it is in a feature freeze or RC phase

> * `PR target: lts`: A criticial fix for a specific release-train that is still within the long term support phase

Reviewer: I think as per document & changes I did, we changed the format to `target: <..>` (note the omitted "PR"), right?

Also it might be worth capturing:

* PRs using `target: lts` need to select the LTS branch manually through the Github UI
* If `target: patch` or `target: rc` are selected, and the PR targets the patch/RC branch, then the PR is not cherry-picked but directly merged into the target branch. like patch-only or rc-only.

To communicate the target we use the following labels:

> Targetting active release train:

Reviewer 2: ```suggestion
Targeting an active release train:
```

> * `target: major`: A breaking change that cannot be part of a minor or patch release
> * `target: minor`: A new feature that cannot be part of a patch release
> * `target: patch`: A bug fix, refactoring, documentation change, etc. that can be a part of a patch release

Reviewer 2: I might avoid the circular references here (e.g. a patch change is a change that can go in a patch release)
```suggestion
* `target: major`: Any breaking change
* `target: minor`: Any new feature
* `target: patch`: Bug fixes, refactorings, documentation changes, etc. that pose no or very low risk of adversely affecting existing applications.
```

Special Cases:
* `target: rc`: A critical fix for an active release-train while it is in a feature freeze or RC phase
* `target: lts`: A criticial fix for a specific release-train that is still within the long term support phase

Notes:
  - To land a change only in a patch/RC branch, without landing it in the latest release-train branch as well,

>   the patch/RC branch can be targetted in the Github UI with the appropriate `target: patch`/`target: rc` label.

Reviewer 2: ```suggestion
  the patch/RC branch can be targeted in the Github UI with the appropriate `target: patch`/`target: rc` label.
```
targetted is the UK spelling

To communicate the target we use the following labels:

Targeting an active release train:

* `target: major`: Any breaking change
* `target: minor`: Any new feature
* `target: patch`: Bug fixes, refactorings, documentation changes, etc. that pose no or very low risk of adversely
  affecting existing applications.

Special Cases:
* `target: rc`: A critical fix for an active release-train while it is in a feature freeze or RC phase
* `target: lts`: A criticial fix for a specific release-train that is still within the long term support phase

Notes:

>   - To land a change only in a patch/RC branch, without landing it in the latest release-train branch as well,

Reviewer: Might be more clearer to say `without landing it in any other active release-train branch (such as `master`). This is slightly more correct as `target: rc` does not cherry-pick into the `next `release-train (master) and into the latest release-train (i.e. patch branch)
